chapter04/rpi/ex3.pm

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+$all = {
```

Comment:
Guess you didn't run either of these, ``strict`` wouldn't have let you off with this.